clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -720,14 +729,26 @@ public FetchResponseMetricAggregator(FetchManagerMetrics sensors,
          * and number of records parsed. After all partitions have reported, we write the metric.
          */
         public void record(TopicPartition partition, int bytes, int records) {
-            unrecordedPartitions.remove(partition);
-            totalBytes += bytes;
-            totalRecords += records;
+            this.unrecordedPartitions.remove(partition);
+            this.fetchMetrics.increment(bytes, records);
 
-            if (unrecordedPartitions.isEmpty()) {
+            // collect and aggregate per-topic metrics
+            String topic = partition.topic();
+            if (!this.topicFetchMetrics.containsKey(topic))
```

Comment:
This is a micro-optimization, so feel free to ignore. An alternative idiom is to call `map.get()` and check the result against null. This works because we know the value will never be null and it saves a hash lookup.